web/ui/react-app/src/Graph.tsx

```diff
@@ -28,11 +37,20 @@ interface GraphProps {
     resolution: number;
   } | null;
 }
+interface GraphState {
+  selectedSeriesIndex: number;
```

Comment:
It'd seem cleaner to me to use `number | null` or an optional property than using a special sentinel value `-1` of the same numeric domain for checking whether there is a series index.